src/main/java/org/cactoos/codec/Base64ToInput.java

```diff
@@ -0,0 +1,106 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.codec;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Base64;
+import org.cactoos.Input;
+import org.cactoos.text.BytesAsText;
+import org.cactoos.text.FormattedText;
+import org.cactoos.text.UncheckedText;
+
+/**
+ * Base64 to Input.
+ * <p>
+ * <p>The class is immutable and thread-safe.
+ *
+ * @author Mehmet Yildirim (memoyil@gmail.com)
+ * @version $Id$
+ * @since 0.12
+ */
+public final class Base64ToInput implements Input {
+
+    /**
+     * All legal Base64 chars.
+     */
+    private static final String BASE64CHARS =
+        "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/=";
+
+    /**
+     * Original input.
+     */
+    private final Input origin;
+
+    /**
+     * Ctor.
+     *
+     * @param input Original
+     */
+    public Base64ToInput(final Input input) {
+        this.origin = input;
+    }
+
+    @Override
+    public InputStream stream() throws IOException {
+        final InputStream stream = this.origin.stream();
+        final ByteArrayOutputStream out = new ByteArrayOutputStream();
+        int read = stream.read();
+        while (read != -1) {
+            out.write(read);
+            read = stream.read();
+        }
+        final byte[] bytes = out.toByteArray();
```

Comment:
@memoyil you seem to be reimplementing what we already have in `InputAsBytes`. No need to read it here, just use `new InputAsBytes(this.origin).bytes()`